src/main/scala/mesosphere/marathon/core/matcher/reconcile/impl/OfferMatcherReconciler.scala

```diff
@@ -37,60 +35,54 @@ private[reconcile] class OfferMatcherReconciler(instanceTracker: InstanceTracker
 
   override def matchOffer(offer: Offer): Future[MatchedInstanceOps] = {
 
-    val frameworkId = FrameworkId("").mergeFromProto(offer.getFrameworkId)
-
-    val resourcesByInstanceId: Map[Instance.Id, Seq[Resource]] = {
-      // TODO(PODS): don't use resident resources yet. Once they're needed it's not clear whether the labels
-      // will continue to be task IDs, or pod instance IDs
-      offer.getResourcesList.groupBy(TaskLabels.instanceIdForResource(frameworkId, _)).collect {
-        case (Some(instanceId), resources) => instanceId -> resources.to[Seq]
+    val resourcesByReservationId: Map[Reservation.Id, Seq[Resource]] = {
+      offer.getResourcesList.groupBy(TaskLabels.reservationFromResource(_)).collect {
+        case (Some(reservationId), resources) => reservationId -> resources.to[Seq]
       }
     }
 
-    processResourcesByInstanceId(offer, resourcesByInstanceId)
+    processResourcesByReservationId(offer, resourcesByReservationId)
   }
 
   /**
     * Generate auxiliary instance operations for an offer based on current instance status.
     * For example, if an instance is no longer required then any resident resources it's using should be released.
     */
-  private[this] def processResourcesByInstanceId(
-    offer: Offer, resourcesByInstanceId: Map[Instance.Id, Seq[Resource]]): Future[MatchedInstanceOps] =
+  private[this] def processResourcesByReservationId(
+    offer: Offer, resourcesByReservationId: Map[Reservation.Id, Seq[Resource]]): Future[MatchedInstanceOps] =
     {
       // do not query instanceTracker in the common case
-      if (resourcesByInstanceId.isEmpty) Future.successful(MatchedInstanceOps.noMatch(offer.getId))
+      if (resourcesByReservationId.isEmpty) Future.successful(MatchedInstanceOps.noMatch(offer.getId))
       else {
-        def createInstanceOps(instancesBySpec: InstancesBySpec, rootGroup: RootGroup): MatchedInstanceOps = {
-
-          /* Was this task launched from a previous app definition, or a prior launch that did not clean up properly */
-          def spurious(instanceId: Instance.Id): Boolean =
-            instancesBySpec.instance(instanceId).isEmpty ||
-              (rootGroup.app(instanceId.runSpecId).isEmpty && rootGroup.pod(instanceId.runSpecId).isEmpty)
-
-          val instanceOps: Seq[InstanceOpWithSource] = resourcesByInstanceId.collect {
-            case (instanceId, spuriousResources) if spurious(instanceId) =>
-              val unreserveAndDestroy =
-                InstanceOp.UnreserveAndDestroyVolumes(
-                  stateOp = InstanceUpdateOperation.ForceExpunge(instanceId),
```

Comment:
Fun fact: This removes another internal usage of `ForceExpunge`! Only the `ExpungeOverdueLostTasksActorLogic`, `TaskKiller` and `OfferProcessorImpl#revertTaskOp` use the force expunge via the `InstanceTrackerDelegate`. I'm pretty sure we could remove the other internal usages as well.

Also, it is kind of funny that we force expunge even though we assume the instance if gone.